src/routes/safe/components/Transactions/helpers/Simulation/TxSimulation.tsx

```diff
@@ -0,0 +1,119 @@
+import { BaseTransaction } from '@gnosis.pm/safe-apps-sdk'
+import styled from 'styled-components'
+
+import { Accordion, AccordionSummary, Loader, Text } from '@gnosis.pm/safe-react-components'
+import { Button } from '@material-ui/core'
+import { FETCH_STATUS } from 'src/utils/requests'
+import { isSimulationAvailable } from './simulation'
+import { SimulationResult } from './SimulationResult'
+import { useSimulation } from './useSimulation'
+import Track from 'src/components/Track'
+import { MODALS_EVENTS } from 'src/utils/events/modals'
+import { useSelector } from 'react-redux'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { getWeb3 } from 'src/logic/wallets/getWeb3'
+
+const StyledAccordionSummary = styled(AccordionSummary)`
+  & .MuiAccordionSummary-content {
+    justify-content: space-between;
+    align-items: center;
+  }
+
+  & .MuiAccordionSummary-expandIcon {
+    display: none;
+  }
+`
+
+const StyledResultAccordionSummary = styled(AccordionSummary)`
+  padding: 0px;
+  & .MuiAccordionSummary-content {
+    justify-content: space-between;
+    align-items: center;
+    padding: 0px;
+  }
+
+  & .MuiAccordionSummary-expandIcon {
+    display: none;
+  }
+`
+
+const LoaderText = styled.span`
+  margin-left: 10px;
+`
+
+type TxSimulationProps = {
+  tx: Omit<BaseTransaction, 'value'>
+  canTxExecute: boolean
+  gasLimit?: string
+  disabled: boolean
+}
+
+export const TxSimulation = ({
+  tx,
+  canTxExecute,
+  gasLimit,
+  disabled,
+}: TxSimulationProps): React.ReactElement | null => {
+  const { simulateTransaction, simulation, simulationRequestStatus, simulationLink, requestError, resetSimulation } =
+    useSimulation()
+  const { chainId, address: safeAddress } = useSelector(currentSafe)
+  const userAddress = useSelector(userAccountSelector)
+  const web3 = getWeb3()
+
+  const getBlockGasLimit = async () => {
```

Comment:
i'd put this in useEffect and fetch it only once

Reply:
I wrapped it in `useAsync()`